src/tests/Microsoft.Diagnostics.NETCore.Client/TestRunner.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using Xunit;
+using Xunit.Abstractions;
+
+using Microsoft.Diagnostics.TestHelpers;
+
+using Microsoft.Diagnostics.NETCore.Client;
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    public class TestRunner
+    {
+        private Process testProcess;
+        private ProcessStartInfo startInfo;
+        private ITestOutputHelper outputHelper;
+
+        public TestRunner(string testExePath, ITestOutputHelper _outputHelper=null)
+        {
+            startInfo = new ProcessStartInfo(CommonHelper.HostExe, testExePath);
+            startInfo.UseShellExecute = false;
+            startInfo.RedirectStandardOutput = true;
+            outputHelper = _outputHelper;
+        }
+
+        public void AddEnvVar(string key, string value)
+        {
+            startInfo.EnvironmentVariables[key] = value;
+        }
+
+        public void Start(int timeoutInMS=0)
+        {
+            if (outputHelper != null)
+                outputHelper.WriteLine("$[{DateTime.Now.ToString()}] Launching test: " + startInfo.FileName);
+
+            testProcess = Process.Start(startInfo);
+
+            if (testProcess == null)
+            {
+                outputHelper.WriteLine($"Could not start process: " + startInfo.FileName);
+            }
+
+            if (testProcess.HasExited)
+            {
+                outputHelper.WriteLine($"Process " + startInfo.FileName + " came back as exited");
+            }
+
+            if (outputHelper != null)
+            {
+                outputHelper.WriteLine($"[{DateTime.Now.ToString()}] Successfuly started process {testProcess.Id}");
+                outputHelper.WriteLine($"Have total {testProcess.Modules.Count} modules loaded");
+            }
+
+            outputHelper.WriteLine($"[{DateTime.Now.ToString()}] Sleeping for {timeoutInMS} ms.");
+            Thread.Sleep(timeoutInMS);
```

Comment:
Sleeping in tests is unreliable on slow machines and can become very wasteful on fast machines. Initially it seems like a few seconds won't matter, but it really can add up as people clone the pattern into more tests, run the same tests over and over in different configurations, and keep increasing the sleep times as quick workarounds for tests that fail on incredibly slow machines.

If the goal is to wait for a the domain socket to be created then I recommend polling for it at a fairly short interval such as every 50ms. If it hasn't been created within the timeout period then trigger an error from the test and exit.